tsdb/engine/tsm1/engine_test.go

```diff
@@ -2701,6 +2702,8 @@ func NewEngine(index string) (*Engine, error) {
 	// store level.
 	seriesIDs := tsdb.NewSeriesIDSet()
 	opt.SeriesIDSets = seriesIDSets([]*tsdb.SeriesIDSet{seriesIDs})
+	// For testing purposes lets set a compaction limiter to 20
+	opt.CompactionLimiter = limiter.NewFixed(20)
```

Comment:
Had to add a compaction limiter in order to actually test SmartOptimize plan type